test-data/unit/fine-grained.test

```diff
@@ -1796,3 +1796,51 @@ def foo(x: Point) -> int:
 [out]
 ==
 b.py:3: error: Unsupported operand types for + ("int" and "str")
+
+[case testNonePartialType]
+import a
+a.y
+
+x = None
+
+def f() -> None:
+    global x
+    x = 1
+[file a.py]
+y = 0
+[file a.py.2]
+y = ''
+[out]
+main:4: error: Need type annotation for 'x'
+==
+main:4: error: Need type annotation for 'x'
+
+[case testNonePartialType2]
+import a
+a.y
+
+x = None
+
+def f():
+    global x
+    x = 1
+[file a.py]
+y = 0
+[file a.py.2]
+y = ''
+[out]
+main:4: error: Need type annotation for 'x'
+==
+main:4: error: Need type annotation for 'x'
+
+[case testNonePartialType3]
+import a
+[file a.py]
+[file a.py.2]
+y = None
+def f() -> None:
+    global y
+    y = ''
+[out]
+==
+a.py:1: error: Need type annotation for 'y'
```

Comment:
Just for completes I would add a tests where the error appears in the first run, but not in the second one (like if a user listened and actually added an annotation).

Reply:
Added a test case.